config/initializers/secure_headers.rb

```diff
@@ -68,3 +68,39 @@
   #   ]
   # }
 end
+
+# A tiny middleware that calls a block on each request.
+# If the block returns true, it deletes the Content-Security-Policy header
+# Intended so that we can override SecureHeaders behavior and not set
+# the headers on asset files
+class SecureHeaders::RemoveContentSecurityPolicy
+  # @yieldparam [Rack::Request] request
+  def initialize(app, &block)
+    @app = app
+    @block = block
+  end
+
+  def call(env)
+    status, headers, body = @app.call(env)
+
+    should_remove = @block.call(Rack::Request.new(env))
+    headers.delete('Content-Security-Policy') if should_remove
+
+    [status, headers, body]
+  end
+end
+
+# We need this to be called after the SecureHeaders::Railtie adds its own middleware at the top
+Rails.application.configure do |config|
+  acuant_sdk_static_files = %w[
+    AcuantImageProcessingWorker.min.js
+    AcuantImageProcessingWorker.wasm
+  ].freeze
+
+  config.middleware.insert_before(
+    SecureHeaders::Middleware,
+    SecureHeaders::RemoveContentSecurityPolicy,
+  ) do |request|
+    acuant_sdk_static_files.any? { |file| request.path.end_with?(file) }
```

Comment:
Given the nature of this logic, should we be a bit more precise about testing that the request path is strictly equal to a full SDK static file path?

To avoid a situation where we have a route parameter that accepts arbitrary user-provided values, where we could request `/path/to/whatever/AcuantImageProcessingWorker.wasm`. Or even the 404 `/nonsense/AcuantImageProcessingWorker.wasm` should still have the CSP headers.

Reply:
One issue is that we have multiple permutations of the asset path, and I'd like to minimize the number of places we list them (check the spec file for the `capture_doc` `capture-doc` `doc_auth` loops)

In a previous iteration, I had logic that checked for a 200-level request, so we would not delete the headers from a 404 page. If I brought that back, would it address your concerns?

Reply:
Based on Slack discussion that we no longer need relative paths, updated in c644dbc0498f3e555665f7020806ec6d0566ceee

Reply:
> One issue is that we have multiple permutations of the asset path, and I'd like to minimize the number of places we list them (check the spec file for the `capture_doc` `capture-doc` `doc_auth` loops)

Possibility here: We might not need to worry about `capture-doc`, since AFAIK it's only used for the initial redirect during a hybrid flow, but ultimately the user should arrive at `capture_doc`.

https://github.com/18F/identity-idp/blob/298f8f928c8ec41c60e8bd4c950cae2603224267/config/routes.rb#L320-L322

>In a previous iteration, I had logic that checked for a 200-level request, so we would not delete the headers from a 404 page. If I brought that back, would it address your concerns?

I think that solves the 404 issue specifically, and at least any potential attack vector I could come up with based on routes we _currently_ have, though it leaves me a bit uneasy that it'd be as easy as introducing a `get '/foo/:param'` without 404-validation on that `:param` to drop CSP headers from a request.

Reply:
- Ok I'll remove `capture-doc` for now, and will make sure to test the SMS flow when we deploy this  (912383387f1480888e50d6ffd64e044ab424fee3)
- Went ahead and explicitly listed all the asset URLs (a5e804ec42be3646e4bf7994aa7664cfb99f1806)